test/distributed/fsdp/test_fsdp_ignored_modules.py

```diff
@@ -146,7 +146,15 @@ def _test_ignored_modules_transformer(
         with FSDP.summon_full_params(wrapped_model):
             flat_param = wrapped_model.params[0]
             flat_param_numel = flat_param.numel()
-            self.assertEqual(flat_param_numel, nonignored_numel)
+            if use_orig_params:
+                # Subtract numel from alignment padding
+                padding_numel = sum(
+                    numel
+                    for (numel, pi) in zip(flat_param._numels, flat_param._param_infos)
+                    if pi is None
```

Comment:
could we ever have `None` in _param_infos if it's not a padding phantom parameter? This might be a bit risky, could we have some global constant to indicate this is a padding parameter and not a real parameter?

Reply:
I refactored the PR so that we have two versions of each data structure that can contain a padding element. This means that the developer will need to make an explicit decision which version to use. Hopefully this helps.